apps/server/src/modules/user-login-migration/service/migration-check.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { SchoolDO } from '@shared/domain/domainobject/school.do';
+import { UserDO } from '@shared/domain/domainobject/user.do';
+import { SchoolService } from '@src/modules/school';
+import { UserService } from '@src/modules/user';
+
+@Injectable()
+export class MigrationCheckService {
+	constructor(private readonly userService: UserService, private readonly schoolService: SchoolService) {}
+
+	async shouldUserMigrate(externalUserId: string, systemId: EntityId, officialSchoolNumber: string): Promise<boolean> {
+		const school: SchoolDO | null = await this.schoolService.getSchoolBySchoolNumber(officialSchoolNumber);
+		const user: UserDO | null = await this.userService.findByExternalId(externalUserId, systemId);
```

Comment:
you could move this between line 16/17 as it is only used inside that if bracket